pandas/tests/series/indexing/test_alter_index.py

```diff
@@ -512,5 +512,28 @@ def test_drop():
 
     # GH 16877
     s = Series([2, 3], index=[0, 1])
-    with tm.assert_raises_regex(KeyError, 'not contained in axis'):
+    with tm.assert_raises_regex(KeyError, 'not found in axis'):
         s.drop([False, True])
+
+
+@pytest.mark.parametrize('index, drop_labels', [
+    ([1, 2, 3], []),
+    ([1, 1, 2], []),
+    ([1, 2, 3], [2]),
+    ([1, 1, 3], [1]),
+])
+def test_drop_empty_list(index, drop_labels):
+    # GH 21494
+    expected_index = [i for i in index if i not in drop_labels]
+    df = pd.DataFrame(index=index).drop(drop_labels)
+    assert (df.index.values == expected_index).all()
```

Comment:
alway do a ``tm.assert_*`` comparison. 

this test is in ``/series/`` but you are testing ``/dataframe`` can you move. its ok to leave if you make this a series (and add to ``/frame/`` as well)